Wikipedia/Code/TalkPageViewModel.swift

```diff
@@ -78,17 +79,21 @@ final class TalkPageViewModel {
             }
         }
     }
-    
+
     func postTopic(topicTitle: String, topicBody: String, completion: @escaping(Result<Void, Error>) -> Void) {
         dataController.postTopic(topicTitle: topicTitle, topicBody: topicBody, completion: completion)
     }
     
+    func postReply(commentId: String, comment: String, completion: @escaping(Result<Void, Error>) -> Void) {
+        dataController.postReply(commentId: commentId, comment: comment, completion: completion)
+    }
+    
 
-    func updateSubscriptionToTopic(topic: String, shouldSubscribe: Bool, completion: @escaping (Result<Bool, Error>) -> Void) {
+    func subscribe(to topic: String, shouldSubscribe: Bool, completion: @escaping (Result<Bool, Error>) -> Void) {
         dataController.subscribeToTopic(topicName: topic, shouldSubscribe: shouldSubscribe) { [self] result in
             switch result {
             case let .success(result) :
-                let topicUpdated = topics.filter { $0.topicTitle == topic}
+                let topicUpdated = topics.filter { $0.topicName == topic}
                 topicUpdated[0].isSubscribed = result
```

Comment:
Let's not actually flip the flag here, and instead do it optimistically before calling subscribe and update the cell state (see comment above). So these lines can be removed.